drivers/Aqara/aqara-lock/src/init.lua

```diff
@@ -0,0 +1,233 @@
+local security = require "st.security"
+local ZigbeeDriver = require "st.zigbee"
+local data_types = require "st.zigbee.data_types"
+local cluster_base = require "st.zigbee.cluster_base"
+local capabilities = require "st.capabilities"
+local base64 = require "base64"
+local credential_utils = require "credential_utils"
+
+local remoteControlStatus = capabilities.remoteControlStatus
+local lockCredentialInfo = capabilities["stse.lockCredentialInfo"]
+local Battery = capabilities.battery
+local Lock = capabilities.lock
+local TamperAlert = capabilities.tamperAlert
+
+local PRI_CLU = 0xFCC0
+local PRI_ATTR = 0xFFF3
+local MFG_CODE = 0x115F
+
+local serial_num = 0
+local seq_num = 0
+
+local SHARED_KEY = "__shared_key"
+local CLOUD_PUBLIC_KEY = "__cloud_public_key"
+
+local function my_secret_data_handler(driver, device, secret_info)
+  -- At time of writing this returns nothind beyond "secret_type = aqara"
```

Comment:
Even though `secret_info.secret_type` is always going to be `aqara` it would be better to check in this function to make sure of that and ignore the event if it doesn't match `aqara`. Future updates to the library might include new messages where `secret_type` will be different and include different payloads. Those changes would cause this driver to crash because of library changes.

Reply:
Current "secret_info.secret_type" is nil. I will add the code when the "secret_info.secret_type" value is updated.
```
2024-04-03T13:52:52.466126870+09:00 PRINT Aqara Lock  ----- [my_secret_data_handler] secret_info.secret_type = nil
```
```lua
local function my_secret_data_handler(driver, device, secret_info)
  print(string.format("----- [my_secret_data_handler] secret_info.secret_type = %s", secret_info.secret_type))
  local shared_key = secret_info.shared_key
  local cloud_public_key = secret_info.cloud_public_key

  device:set_field(SHARED_KEY, shared_key, { persist = true })
  device:set_field(CLOUD_PUBLIC_KEY, cloud_public_key, { persist = true })
  credential_utils.save_data(driver)

  if cloud_public_key ~= nil then
    local raw_data = base64.decode(cloud_public_key)
    -- send cloud_pub_key
    device:send(cluster_base.write_manufacturer_specific_attribute(device,
      PRI_CLU, PRI_ATTR, MFG_CODE, data_types.OctetString, "\x3E" .. raw_data))
  end
end
```

Reply:
Oh this is my mistake. I think the comment is from an older version of the libraries. The correct key is `secret_kind` - I had originally written it as `secret_type` at one point and it must have made it into some reference or something.